src/main/java/client/SkillFactory.java

```diff
@@ -38,25 +38,26 @@ public static Skill getSkill(int id) {
         return skills.get(id);
     }
 
-    public static void loadAllSkills() {
-        final Map<Integer, Skill> loadedSkills = new HashMap<>();
-        final DataDirectoryEntry root = datasource.getRoot();
-        for (DataFileEntry topDir : root.getFiles()) { // Loop thru jobs
-            if (topDir.getName().length() <= 8) {
-                for (Data data : datasource.getData(topDir.getName())) { // Loop thru each jobs
-                    if (data.getName().equals("skill")) {
-                        for (Data data2 : data) { // Loop thru each jobs
-                            if (data2 != null) {
-                                int skillId = Integer.parseInt(data2.getName());
-                                loadedSkills.put(skillId, loadFromData(skillId, data2));
-                            }
-                        }
-                    }
+    ppublic static void loadAllSkills() {
+    final Map<Integer, Skill> loadedSkills = new HashMap<>();
+    final DataProvider dataProvider = DataProviderFactory.getDataProvider(WZFiles.SKILL);
```

Comment:
You didn't remove the original dataprovider at the top of the class, so you're just duplicating all the junk being ran in this constructor